src/asami/entities.cljc

```diff
@@ -130,7 +142,8 @@
    data :- [s/Any]]
   (let [graph (storage/graph db)
         [retract-stmts new-data] (util/divide' #(= :db/retract (first %)) data)
-        retractions (mapv #(subvec % 1 4) retract-stmts)
+        ref->id (partial resolve-lookup-refs graph)
+        retractions (mapv #(into [(ref->id (second %))] (subvec % 2 4)) retract-stmts)
```

Comment:
Perhaps it should consider the values as well?
```clojure
retractions (mapv (fn [_ e a v] [(ref->id e) a (ref->id v)]) retract-stmts)
```

Reply:
Or...
We have a use case where people want to use objects as attributes. The modeling for this twists my brain, but there's no reason it can't be done. So to cover this case, I suggest:
```clojure
retractions (mapv (partial mapv ref-id) retract-stmts)
```

Reply:
I hadn't had coffee, and made a mistake here (sorry)
It should have been:
```clojure
retractions (mapv (comp (partial mapv ref-id) rest) retract-stmts)
```